### ISSUE TITLE
Column map expectations (and perhaps others?) sometimes fail on categorical variables

### ISSUE THREAD

**Issue by abegong** [fivetran/great_expectations#374]
Setup:
```
import pandas as pd
import great_expectations as ge
df = ge.dataset.PandasDataset(pd.DataFrame({
    "x" : ["A", "A", "A", "B", "B", "B"]
}))
df.x = df["x"].astype('category')
```

Running: `df.expect_column_values_to_be_in_set("x", ["A", "B"])` successfully returns:
```
{'success': True,
 'result': {'element_count': 6,
  'missing_count': 0,
  'missing_percent': 0.0,
  'unexpected_count': 0,
  'unexpected_percent': 0.0,
  'unexpected_percent_nonmissing': 0.0,
  'partial_unexpected_list': []}}
```

However, `df.expect_column_values_to_be_in_set("x", ["Z", "B"])` errors out with:
```
TypeError                                 Traceback (most recent call last)
<ipython-input-23-e098449c62a6> in <module>()
----> 1 df.expect_column_values_to_be_in_set("x", ["Z", "B"])

~/anaconda2/envs/py36/lib/python3.6/site-packages/great_expectations/dataset/util.py in f(*args, **kwargs)
     98         @wraps(self.mthd, assigned=('__name__', '__module__'))
     99         def f(*args, **kwargs):
--> 100             return self.mthd(obj, *args, **kwargs)
    101 
    102         f.__doc__ = doc

~/anaconda2/envs/py36/lib/python3.6/site-packages/great_expectations/dataset/base.py in wrapper(self, *args, **kwargs)
    139 
    140                     else:
--> 141                         raise(err)
    142 
    143                 # Append the expectation to the config.

~/anaconda2/envs/py36/lib/python3.6/site-packages/great_expectations/dataset/base.py in wrapper(self, *args, **kwargs)
    126                 # Finally, execute the expectation method itself
    127                 try:
--> 128                     return_obj = func(self, **evaluation_args)
    129 
    130                 except Exception as err:

~/anaconda2/envs/py36/lib/python3.6/site-packages/great_expectations/dataset/pandas_dataset.py in inner_wrapper(self, column, mostly, result_format, *args, **kwargs)
     80 
     81             boolean_mapped_success_values = func(self, nonnull_values, *args, **kwargs)
---> 82             success_count = boolean_mapped_success_values.sum()
     83 
     84             unexpected_list = list(nonnull_values[boolean_mapped_success_values==False])

~/anaconda2/envs/py36/lib/python3.6/site-packages/pandas/core/generic.py in stat_func(self, axis, skipna, level, numeric_only, min_count, **kwargs)
   7293                                       skipna=skipna, min_count=min_count)
   7294         return self._reduce(f, name, axis=axis, skipna=skipna,
-> 7295                             numeric_only=numeric_only, min_count=min_count)
   7296 
   7297     return set_function_name(stat_func, name, cls)

~/anaconda2/envs/py36/lib/python3.6/site-packages/pandas/core/series.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   2579         return delegate._reduce(op=op, name=name, axis=axis, skipna=skipna,
   2580                                 numeric_only=numeric_only,
-> 2581                                 filter_type=filter_type, **kwds)
   2582 
   2583     def _reindex_indexer(self, new_index, indexer, copy):

~/anaconda2/envs/py36/lib/python3.6/site-packages/pandas/core/categorical.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   1963         if func is None:
   1964             msg = 'Categorical cannot perform the operation {op}'
-> 1965             raise TypeError(msg.format(op=name))
   1966         return func(numeric_only=numeric_only, **kwds)
   1967 

TypeError: Categorical cannot perform the operation sum
```

**Comment by eugmandel** [comment]
I think that just like in this PR https://github.com/great-expectations/great_expectations/pull/367/ the fix would be changing how successes are counted - replacing .sum on a categorical column (which cannot run because category is not a numpy dtype) with np.count_nonzero.

**Comment by abegong** [comment]
Care to submit a PR?

**Comment by elsander** [comment]
I can take a stab at this! (I'm taking a hack week at work to contribute to the library, which is why you're getting so many PRs from me all at once)

**Comment by elsander** [comment]
I can't reproduce this. Did a fix already get merged in?

**Comment by RoyalTS** [comment]
Is this still an issue? Where would I put a test for this behavior and find out?

**Comment by jcampbell** [comment]
Thanks for flagging this as stale...yes a fix was merged in in #367 I'll close the issue.
